### PR TITLE
Fix set_value clds spin

### PR DESCRIPTION
**Pull request by dcristoloveanu**
Fix set_value clds spin

### CODE REVIEW

File: src/clds_sorted_list.c [hunk 1]
```diff
@@ -769,8 +769,11 @@ CLDS_SORTED_LIST_INSERT_RESULT clds_sorted_list_insert(CLDS_SORTED_LIST_HANDLE c
                 // get the current_item value
                 volatile CLDS_SORTED_LIST_ITEM* current_item = (volatile CLDS_SORTED_LIST_ITEM*)InterlockedCompareExchangePointer((volatile PVOID*)current_item_address, NULL, NULL);
 
+                // clear any delete lock bit from what we read
+                current_item = (volatile CLDS_SORTED_LIST_ITEM*)((uintptr_t)current_item & ~0x1);
```
Comment (mattdurak) [inline src/clds_sorted_list.c:773]: >current_item = (volatile CLDS_SORTED_LIST_ITEM*)((uintptr_t)current_item & ~0x1); [](start = 16, length = 81)

I kind of feel like this should be wrapped in some kind of handle/macros to make it less likely to miss...

Ideally, the "handle" would expose something like set/clear bit and "get pointer without delete bit" and somehow fail if it's used incorrectly.

maybe not worth doing right now but could be a task for the backlog. #WontFix

**Comment by mattdurak** [comment]
>         CHAOS_TEST_ACTION action = (CHAOS_TEST_ACTION)(rand() * ((MU_COUNT_ARG(CHAOS_TEST_ACTION_VALUES)) - 1) / RAND_MAX);

Should be divided by `RAND_MAX + 1`, otherwise this selects an invalid value when rand() == RAND_MAX.

---
Refers to: tests/clds_sorted_list_int/clds_sorted_list_int.c:1287 in 7818b62. [](commit_id = 7818b62caf55b89a80ed1950cff4d7a7a12862c3, deletion_comment = False)

**Review by mattdurak — APPROVED** [review]
:shipit:

File: src/clds_sorted_list.c (hunk 1 above)
Comment (dcristoloveanu) [reply src/clds_sorted_list.c:773]: Yeah, I guess at a later time ....

---
In reply to: [421122049](https://github.com/dcristoloveanu/clds/pull/47#discussion_r421122049) [](ancestors = 421122049)

**Pull request merged**